src/benchmark/backendbench.cc

```diff
@@ -123,15 +117,15 @@ void BackendBenchmark::Run() {
 
       if (option_dict.Get<bool>(kClippyId)) {
         const float threshold = option_dict.Get<float>(kClippyThresholdId);
-        const float tolerance = option_dict.Get<float>(kClippyToleranceId);
-
-        if (nps > best_nps &&
-            (run == true || nps > best_nps * (1.0f + threshold))) {
-          best_nps = nps;
-          best = i;
-          run = true;
-          if (!pending) {
-            pending = std::chrono::steady_clock::now();
+
+        if (nps > best_nps) {
+          if (best==0 ||
+             ((i*1.0f/best*1.0f-1.0f)*threshold < nps/best_nps-1.0f)) {
```

Comment:
You can rewrite this as `(i-best)*threshold*best_nps < (nps-best_nps)*best`.
Then you can initialize best to 1 to avoid the best==0 check and the nested `if`.